src/resolvelib/providers.py

```diff
@@ -9,14 +9,7 @@ def identify(self, requirement_or_candidate):
         """
         raise NotImplementedError
 
-    def get_preference(
-        self,
-        identifier,
-        resolutions,
-        candidates,
-        information,
-        backtrack_causes,
-    ):
+    def get_preference(self, identifier, resolutions, candidates, information):
```

Comment:
We’re at version 1 now so it’d be better if we could implement this in a backward-compatible way. Maybe we can still pass the backtrack causes anyway?

Reply:
Sure, it was just to simplify the interface and make sure downstream libraries aren't duplicating this logic, but I don't have any strong opinions on whether to keep providing it or not.